### PR TITLE
Notify callback

### LINKED ISSUE #47: Support a callback URI
After a task is completed or failed, POST exit-status to a callback URI

### PR DESCRIPTION
**Pull request by alde**
When IsTerminal is reached, notify a callback by sending the
final State with a HTTP POST.

Fixes #47

### CODE REVIEW

File: handler/handler.go [hunk 1]
```diff
@@ -60,6 +61,29 @@ func GetTaskInfo(scheduler types.Scheduler) http.HandlerFunc {
...
+	}
+
+	cbData := task.Status[len(task.Status)-1]
+
+	body, err := json.Marshal(cbData)
+	if err != nil {
+		log.Errorf("Unable to create message for task %s, target uri %s", task.ID, task.CallbackURI)
+	}
+
+	_, err = http.Post(task.CallbackURI, "application/json", bytes.NewBuffer(body))
```
Comment (keis) [inline handler/handler.go]: I guess this is waiting for the response before returning so potentially a really long wait for a time out here. Should we stick this is in a goroutine to not risk blocking things?

Comment (alde) [inline handler/handler.go]: good point, will do

**Comment by keis** [comment]
The callback uri (along with any other metadata) will not survive a full restart of eremetic so the promise of this api becomes a bit weak, "you might get notified when task finishes". I think that can be fine though but it's worth noting and also maybe include in the documentation.

File: README.md [hunk 2]
```diff
@@ -10,6 +10,32 @@ Send a cURL to the eremetic framework with how much cpu and memory you need, wha
...
+        {
+          "container_path": "/var/run/docker.sock",
+          "host_path": "/var/run/docker.sock"
+        }
+      ],
+      // Object, Environment variables to pass to the container
+      "env": {
+        "KEY": "value"
+      },
+      // String, URL to post a callback to. Callback has format: {"time":1451394641,"status":"TASK_FAILED"}
```
Comment (keis) [inline README.md]: Would be nice to include the task_id too so you don't have to use lots of unique callback urls

**Comment by keis** [comment]
[LGTM](http://lgtm.in/i/wDB1)

**Pull request merged**